### PR TITLE
fix: updated design for save as datasource button

### LINKED ISSUE #12842: [Bug]-[60000]:Rest API Under The Authentication tab "Save as datasource" does not look as an Active button
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When the user adds an API to the API pane it is observed that the "Save as datasource" appears next to the field. When the user navigates to Authentication Tab and observes the Save as datasource it is observed that the "Save as datasource" is not displayed as active button 

### Steps To Reproduce

1) Navigate to API Pane 
2) Add a API 
3) Navigate to Authentication Tab
and observe the button it is grey colored and does not seem to be active 
Screenshot 2022-04-12 at 2 55 28 PM

### Public Sample App

_No response_

### Version

Release

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description

Updated the design for `save as datasource` or `edit datasource` button present in Api editor at two places (one in authentication tab and the other on the right side of url field).

### Previous
Screenshot 2022-04-12 at 2 55 28 PM

### Now
Screenshot 2022-09-12 at 6 50 19 PM

Fixes #12842

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/editorComponents/StoreAsDatasource.tsx [hunk 1]
```diff
@@ -32,22 +45,60 @@ const disabled = css`
...
 
+interface ReduxDispatchProps {
+  setDatasourceEditorMode: (id: string, viewMode: boolean) => void;
+}
+
 function StoreAsDatasource(props: storeDataSourceProps) {
   const dispatch = useDispatch();
+  const pageId = useSelector(getCurrentPageId);
+
+  const onClick = () => {
```
Comment (ChandanBalajiBP) [inline app/client/src/components/editorComponents/StoreAsDatasource.tsx:62]: nit: can we rename this function to have better readability?

Comment (ChandanBalajiBP) [inline app/client/src/components/editorComponents/StoreAsDatasource.tsx:62]: nit: can we rename this function to have better readability?

**Review by ChandanBalajiBP — COMMENTED** [review]
Looks good and kudos to you for code refactoring.

File: app/client/src/components/editorComponents/StoreAsDatasource.tsx (hunk 1 above)
Comment (AmanAgarwal041) [reply app/client/src/components/editorComponents/StoreAsDatasource.tsx:62]: Sure, will update the name

**Review by ChandanBalajiBP — APPROVED** [review]

**Pull request merged**